mas-foundation/src/main/java/com/ca/mas/core/storage/sharedstorage/AccountManagerUtil.java

```diff
@@ -51,27 +58,42 @@ public AccountManagerUtil(Context context, String accountName, boolean sharedSto
             mAccountManager = AccountManager.get(MAS.getContext());
             //Attempt to retrieve the account
             Account[] accounts = mAccountManager.getAccountsByType(accountType);
```

Comment:
Apply synchronization to the block that consists of the below logic:
synchronized {
- get-accounts
- iterate
- add-account-explicitly
}